velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -639,4 +639,38 @@ struct ArrayFlattenFunction {
     }
   }
 };
+
+/// This class implements the array remove function.
+///
+/// DEFINITION:
+/// array_remove(x, element) → array
+/// Remove all elements that equal element from array x.
+template <typename TExecCtx, typename T>
+struct ArrayRemoveFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecCtx);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Array<T>>& out,
+      const arg_type<Array<T>>& array,
```

Comment:
lets also add the generic (non-fast path ) support for all types. 
for examples see ArrayTrim function. 

https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/ArrayFunctions.h
you can check the equal function to compare two items or direcly use compare 
(look at eq function here 
https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/Comparisons.h

Reply:
added generic support for all types.

Done.